docs/SQLModelingStandard.md

```diff
@@ -0,0 +1,125 @@
+  - The table name should be in **lowercase**.
+  - The table name made of **multiple word** should use an **underscore** 
+  in order to separate each of them.
+- The table name should be in **singular form**.
+- The table name should contain **English words** only.
+- The table name should **not contain abbreviation** with the exception of 
+  "\_rel" for entity relationship tables.
+- The table name should be a **noun** or **group of nouns**, as concise as 
+  possible.
+- When a **single word is not sufficient** to describe a table, the ordering 
+  of the words in the name should reveal the concern of the table in a 
+  broad-to-specific manner. 
+  - *i.e. A table containing the consent information of a candidate would be 
+    named `candidate_consent`, candidate being the broader concern and just 
+    `consent` is not sufficiently  informative to rule out user consents and 
+    examiner consents. A table containing the consent types would be called 
+    as such `consent_type`.*
+- A table that is **specific to a module** should be **prefixed** with the 
+  module name.
+  - *i.e. A table named `xyz` used in module `abc` should be name `abc_xyz`.*
+- **Acronyms** should be avoided, however may be used when the acronym would 
+  be **commonly used by the neuroimaging community**. The acronym should
+  **only contain upper case letter**. For historic reasons, “PSC” to refer to 
+  a Site is an exception to this rule.
+- When creating a table representing a **relation between two or more tables**. 
+  Regardless of the relation type (many-to-many, one-to-many, many-to-one or 
+  one-to-one), the table should have a composite name as such 
+  `table1_table2_rel`.
+  - *i.e. the table mapping users to their addresses would be named 
+  `user_address_rel` where `user` is the name of the Users' entity table and 
+  `address` is the name of the Addresses' entity table.*
+
+### Fields
+
+- The field name should be in **CapitalizedWords**.
+  - The field name should **start** with an **Upper case letter**.
+  - The field name should **not contain any underscore**.
+- The field name should be in **singular form**.
+- The field name should contain **English words** only.
+- The field name should **not contain abbreviations**.
+- **Acronyms** should be avoided, however may be used when the acronym would 
+  be **commonly used by the neuroimaging community**. The acronym should 
+  **only contain upper case letter**.
+- The **primary key** field of a table should be named `<TableName>ID`. While 
+  the table name is snake_case, the field names referencing it (such as the 
+  primary key) should follow field naming conventions. (ie. `my_table`’s 
+  primary key should be `MyTableID`).
+  - *i.e. the name of the primary identifier of the `country` table should be 
+  `CountryID`.*
+- When an explicit auto-incremented primary key field is present, it should be
+  of type unsigned int. Most tables should have an explicit primary key field.
```

Comment:
All tables **must** have an explicit primary key.

Reply:
But It could be mentioned in the constraints section instead of here

Reply:
This was discussed when it was being worked on in Google docs, and that was weakened because some tables (such as relationship tables) have composite keys.

Reply:
Ok then.

Reply:
But composite keys make perfect primary keys

Reply:
It is not forbidden. But only when appropriate. Again composite key that are too long (not fixed rule but 3 or more fields, or including varchar(20+)) are generally not a good idea.